build.gradle

```diff
@@ -558,7 +558,7 @@ configure(dexerProjects) { subproject ->
     }
 
     afterEvaluate {
-        def checkDexer = task('checkDexer', type: JavaExec, description: 'compiles android class files into dex bytecode', dependsOn: jar) {
+        def checkDexer = task('checkDexer', type: JavaExec, description: 'compiles android class files into dex bytecode', dependsOn: buildAndroidProjects) {
```

Comment:
As said depending on jar should be enough we don't want to execute tests for instance. You need to fix the classpath we pass to dexer. Try gr checkDexer --dry-run I guess everything is already correct

Reply:
Ah no, it must be the other way round, we pass runtime dependencies to dexer but their dependencies aren't build yet. Could be a gradle bug, we already had to do workarounds for it. search for domain-robstollin build.gradle files. Maybe you find the workaround. In any case, perform the dry run and check if atrium-domain-robstoll-android:jar comes earlier than checkDexer (I assume currently it is executed later)

Reply:
```
configure(dexerProjects) {
    compileKotlin.dependsOn(
        prefixedProject('domain-robstoll-android').jar
    )
}
```

Is this the type of change you were thinking of?
Its not working for me but I didnt notice any other special usage of domain-robstoll due to circular dependency

Reply:
Yeah something like that, but I would put it into the project's build.gradle which causes the problem. But no worries, you can also keep it like it is now and I improve it once your PR is merged. For me it's more valuable if you can implement the check that d8 did not output warnings and fail otherwise

Reply:
Added the output verification
Since successful build results in no messages at all, in any case of error message I will fail the build

Reply:
👍